### PR TITLE
Fixes #317 - Logon Entry Batch FIle

### LINKED ISSUE #317: Filthy batch file to add user details to various places
**Is your feature request related to a problem? Please describe.**
User details not entered into various bits of ES...

**Describe the solution you'd like**
batch file to enter the following:
 Insert Hoppie Code to TopSky/vSMR
Insert Name/CID (probably not password) to login screen and VCCS nickname

Or other method that someone else can think of...

### PR DESCRIPTION
**Pull request by SamLefevre**
Fixes #317 

# Summary of changes

Add batch file that automatically adds logon details to profiles and adds CPDLC code to TopSky

# Screenshots (if necessary)

https://i.imgur.com/Z9nPZwl.gif

### CODE REVIEW

**Comment by SamLefevre** [comment]
I will try and find a way to improve it as it is a bit inefficient.

**Comment by SamLefevre** [comment]
Changed to using python as I was struggling to make efficient changes to the batch file, there is a compiled .exe as well as the python file.

**Comment by PLM1995** [comment]
@SamLefevre Out of interest, how easy or hard would it be to add VCCS PTT and Audio Devices into this? I notice you're doing the nickname but not these other bits needed to get it set up for all .prf files all at once. I think this would be really handy if it wasn't too challenging.

Peter.

**Comment by SamLefevre** [comment]
> @SamLefevre Out of interest, how easy or hard would it be to add VCCS PTT and Audio Devices into this? I notice you're doing the nickname but not these other bits needed to get it set up for all .prf files all at once. I think this would be really handy if it wasn't too challenging.
> 
> Peter.

For the PTT I guess they'd have to type it in and as for audio devices, it would have to show a list and allow the user to select one and next write that to whatever file these settings are saved in so I'm not sure I'll be able to do that while keeping the program simple. Not very good at coding, so for that someone else may have to try.

**Comment by PLM1995** [comment]
> > @SamLefevre Out of interest, how easy or hard would it be to add VCCS PTT and Audio Devices into this? I notice you're doing the nickname but not these other bits needed to get it set up for all .prf files all at once. I think this would be really handy if it wasn't too challenging.
> > Peter.
> 
> For the PTT I guess they'd have to type it in and as for audio devices, it would have to show a list and allow the user to select one and next write that to whatever file these settings are saved in so I'm not sure I'll be able to do that while keeping the program simple. Not very good at coding, so for that someone else may have to try.

Fair enough. Perhaps it is beyond the remit of this PR and could be added separately later. I think those settings are saved in plain text in the .prf files. Mine is like this at the bottom of the .prf:

`TeamSpeakVccs	Ts3G2APtt	3670016`
`TeamSpeakVccs	Ts3G2GPtt	3670016`
`TeamSpeakVccs	Ts3NickName	1380981`
`TeamSpeakVccs	PlaybackMode	Windows Audio Session`
`TeamSpeakVccs	PlaybackDevice	Headset Earphone (CORSAIR HS70 Wireless Gaming Headset)`
`TeamSpeakVccs	CaptureMode	Windows Audio Session`
`TeamSpeakVccs	CaptureDevice	Headset Microphone (CORSAIR HS70 Wireless Gaming Headset)`
`TeamSpeakVccs	Ts3DialogLeft	-1229`
`TeamSpeakVccs	Ts3DialogTop	-155`
`TeamSpeakVccs	Ts3DialogRight	-331`
`TeamSpeakVccs	Ts3DialogBottom	445`

Obviously you wouldn't bother doing the position stuff. One very dumb way to do it would be to let the user copy-paste from a .prf in which they had configured the VCCS devices and PTT, like as an optional thing for those who want to? But you're right the proper way to do it would be more complex than that and perhaps this is better left for a possible future update.

---

Is the .bat file still used for anything, or is it just the .py and .exe?

---

I tried running the .exe on a fresh pack download, and hit a few hickups:

- Some profiles work fine (like Glasgow SMR) however some profiles don't input the last session name correctly (like Heathrow SMR). The difference is the Glasgow SMR.prf ends with a blank line, whereas the Heathrow SMR.prf doesn't, so it writes the name at the end of the last line, presumably also breaking the airways file path. I think personally the best bet would be to insert a blank line above the one being added, since some people might run this without downloading a full pack and having to update all the .prfs to be the same at the end would be a faff. Having a blank line between stuff doesn't seem to break anything in the case where one already exists, unless you wanna do something more clever to read the last line and decide. Also, idk if you want to have a blank line at the end after the data you've added, probably doesn't make much difference?

- The area profiles which are 'two folders deep' in the file structure don't seem to get updated at all. I don't know what the problem is, but wonder if using the built in recursion in glob may help and be more future-proofed? I don't know.

- I never get the "... already has details (details have not altered" message, and re-running the .exe continues to add and add the data to the end of the .prf files.

**Comment by SamLefevre** [comment]
I could probably learn how to do the capture device stuff, I've always wanted to learn how to program properly so I could use this as an opportunity. The .py file is the original and then the .exe is the compiled version of the .py. The .bat gives us another option if we don't want to use a .exe file. The .exe I must of uploaded an old version as if you run the .py version it works as it should. The only bug with the .py file is that the Heathrow SMR.prf has the same issue you described and I can easily fix that with adding a line above.

**Comment by SamLefevre** [comment]
image
The plan for the "details have not altered" is that I can change it in the future to overwrite old logon details with new ones.

**Comment by SamLefevre** [comment]
.exe now works and all bugs should be fixed

**Comment by cpawley** [comment]
Can we also save the layout info prior to updating (to retain list positions etc?)

**Comment by luke11brown** [comment]
Any further thoughts on this @KHardern ?

I'm keen to include some sort of solution to allow insertion of the details.

**Comment by mrten64** [comment]
[Tote bag printing](https://dc9.com.sg/singapore-bags-printing/canvas-bag-printing/) also offers sustainability benefits. Tote bags are often made from natural fibers, such as cotton or canvas, which are biodegradable and eco-friendly. By opting for tote bag printing instead of single-use plastic bags, individuals and businesses can contribute to reducing plastic waste and promoting sustainability.

**Comment by SamLefevre** [comment]
I don't think I'll add the VCCS features, better to be done once in EuroScope in my opinion, although could add options for land and coast colour but no more features, that's something that someone on the web team could do far better than I can.

**Comment by luke11brown** [comment]
@KHardern - As the latest version of ES looks to install into appdata, maybe it is now worth considering some sort of installer that will then as part of the process incorporate the functions of this pr?

File: UK/Logon Details.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+# Write CPDLC to TopSky
+
+f = open("Data/Plugin/TopSky/TopSkyCPDLChoppieCode.txt", "w")
+f.write(CPDLC)
+f.close
+
+# Add CID as Nickname to VCCS profiles
+
+PrfVCCS=("TeamSpeakVccs	Ts3NickName	"+CID)
```
[the remark this answers is not in the document]
Comment (SamLefevre) [reply UK/Logon Details.py:26]: EuroScope REQUIRES this EXACT tab otherwise it doesn't work.

**Comment by cpawley** [comment]
> I've had a mess around and got this as a (not yet working) PoC - https://github.com/chssn/ukcp-updater. In essence this will:

That repo is not accessible to me (think it's private)

File: UK/Logon Details.py [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+import glob
+import time
+
+# User Inputs
+
+Name = input("Enter Name: ")
+os.system('cls')
+CID = input("Enter CID: ")
+os.system('cls')
+#Password = input("Enter Password: ")
```
[the remark this answers is not in the document]
Comment (SamLefevre) [reply UK/Logon Details.py:11]: If OPS wants passwords to be saved then we can do that. But it's not necessarily my decision.

**Comment by luke11brown** [comment]
Just testing this evening. Working absolutely grand.

One final feature request. vSMR also has Hoppies capability, the password is stores in various files such as
`UK\Data\Settings\Local\Newcastle\Plugins.txt`

On the line
`vSMR:cpdlc_password:`

Could you also add the Hoppie code in all instances of this?

**Comment by SamLefevre** [comment]
Feature added

**Comment by luke11brown** [comment]
@KHardern Happy for the exe to be included in the next release?

**Review by luke11brown — APPROVED** [review]

**Pull request merged**